### PR TITLE
[OSFUSE-233] Add forge commands to create FIS projects

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Comment by jamesnetherton** [comment]
ok to test

**Comment by jamesnetherton** [comment]
retest this please

File: src/main/java/org/jboss/fuse/forge/addon/completer/ArchetypeVersionCompleter.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    public ArchetypeVersionCompleter(List<String> archetypeVersions) {
+        this.archetypeVersions = archetypeVersions;
+    }
+
+    @Override
+    public Iterable<String> getCompletionProposals(UIContext context, InputComponent<?, String> input, String value) {
+        return archetypeVersions.stream()
+            .filter(a -> a.startsWith(value))
+            .sorted((o1, o2) -> o1.compareTo(o2))
```
Comment (apupier) [inline src/main/java/org/jboss/fuse/forge/addon/completer/ArchetypeVersionCompleter.java]: You can also use Comparator.naturalOrder() which is easier to read.
With String we also often sort ignoring case, you can use String::compareToIgnoreCase

File: src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        JavaSourceFacet facet = project.getFacet(JavaSourceFacet.class);
+        archetypeHelper.setPackageName(facet.getBasePackage());
+        archetypeHelper.execute();
+
+        return Results.success("Created new JBoss Fuse project in: " + fileRoot.getPath());
+    }
+
+    private void configureProjectTypeInput() {
+        Set<String> types = new LinkedHashSet<>();
+        for (FuseProjectCategory category : FuseProjectCategory.values()) {
```
Comment (apupier) [inline src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java]: You can also use map on a stream, you will have somethign like:
Set<String> types = FuseProjectCategory.values().stream()
.map(category -> category.getName())
.collect(Collectors.toCollection(LinkedHashSet::new));

File: src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+        catalogVersion.setRequired(false)
+            .setDefaultValue(getLatestCatalogVersion())
+            .setCompleter(new ArchetypeVersionCompleter(archetypeVersions));
+
+        catalogVersion.addValueChangeListener(valueChangeEvent -> {
+            String version = String.valueOf(valueChangeEvent.getNewValue());
+            archetype.setValueChoices(() -> {
+                Set<Archetype> result = new LinkedHashSet<>();
+                if (isValidVersion(version)) {
+                    ArchetypeCatalog archetypes = getCatalog();
```
Comment (apupier) [inline src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java]: the variable name might be better archetypeCatalog?

File: src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+            return source.getGroupId() + ":" + source.getArtifactId() + ":" + source.getVersion();
+        }).setDescription(() -> {
+            Archetype value = archetype.getValue();
+            return value == null ? null : value.getDescription();
+        });
+
+        ArchetypeCatalog archetypes = getCatalog();
+        if (archetypes != null) {
+            Set<Archetype> result = new LinkedHashSet<>();
+            List<Archetype> list = archetypes.getArchetypes();
```
Comment (apupier) [inline src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java]: list might be better named archetypes?

**Pull request merged**